#! /bin/sh /usr/share/dpatch/dpatch-run
## 23-TTY_IO-compilation-fix.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix compilation of TTY_IO.cpp for platform that don't provide `c_ispeed`
## DP: and `c_ospeed`.

@DPATCH@

--- ACE_wrappers/ace/TTY_IO.cpp-	2005-12-10 12:20:11.000000000 +0100
+++ ACE_wrappers/ace/TTY_IO.cpp	2005-12-10 12:21:00.000000000 +0100
@@ -178,7 +178,7 @@
           return -1;
         }
 
-#if defined(ACE_USES_NEW_TERMIOS_STRUCT)
+#if defined(ACE_USES_NEW_TERMIOS_STRUCT) && defined (_HAVE_STRUCT_TERMIOS_C_ISPEED) && defined (_HAVE_STRUCT_TERMIOS_C_OSPEED)
       // @@ Can you really have different input and output baud
       // rates?!
       devpar.c_ispeed = newbaudrate;
